src/Chapter3.hs

```diff
@@ -971,6 +1074,31 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data DayOfWeek
+    = Monday
+    | Tuesday
+    | Wednesday
+    | Thursday
+    | Friday
+    | Saturday
+    | Sunday
+    deriving (Show, Enum, Eq, Ord)
+
+isWeekend :: DayOfWeek -> Bool
+isWeekend Saturday = True
+isWeekend Sunday = True
+isWeekend _ = False
+
+nextDay :: DayOfWeek -> DayOfWeek
+nextDay day
+  | day == Saturday = Monday
+  | otherwise = succ day
+
+daysToParty :: DayOfWeek -> Int
+daysToParty day
+    | day > Friday = fromEnum Sunday - fromEnum day + fromEnum Friday + 1
+    | otherwise = fromEnum Friday - fromEnum day
```

Comment:
This is correct!
However, this could be a bit simplified and made a bit more independent of the data type structure 👍🏼 

We know for sure that it is 7 days in a week. So we can use `mod` function on the difference, that will tell us exactly how many days left to the Friday:

```haskell
>>> mod (-1) 7
6
```

Reply:
Cool, I've made a note